From: Keith Busch <keith.busch@intel.com>
Subject: nvme: check for disk before submitting cmd
Date: Tue Nov 21 10:59:33 CET 2017
References: bsc#1067848
Patch-mainline: Never, SLE-15 specific

As SLE-15 is missing commit '74d46992e0d9 ("block: replace bi_bdev
with a gendisk pointer and partitions index")' it is still necessary
to check for a NULL disk pointer in nvme_submit_user_cmd() otherwise
we will oops on a NULL pointer dereference when calling bdget_disk().

Signed-off-by: Keith Busch <keith.busch@intel.com>
Acked-by: Johannes Thumshirn <jthumshirn@suse.de>
---
 drivers/nvme/host/core.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/drivers/nvme/host/core.c
+++ b/drivers/nvme/host/core.c
@@ -658,6 +658,8 @@ static int nvme_submit_user_cmd(struct r
 				GFP_KERNEL);
 		if (ret)
 			goto out;
+		if (!disk)
+			goto submit;
 		bio = req->bio;
 		bio->bi_bdev = bdget_disk(disk, 0);
 		if (!bio->bi_bdev) {
@@ -674,7 +676,7 @@ static int nvme_submit_user_cmd(struct r
 			}
 		}
 	}
-
+ submit:
 	blk_execute_rq(req->q, disk, req, 0);
 	if (nvme_req(req)->flags & NVME_REQ_CANCELLED)
 		ret = -EINTR;
